File: fruit.py
class Fruit:
	def __init__(self, data = {"color": "unknown"}):
		self.data = data

	def set_color(self, color: str):
		self.data["color"] = color

	def print_color(self):
		print(f"I'm { self.data.get('color') }")


orange = Fruit()
orange.set_color("orange")
orange.print_color() # Prints "I'm orange"


apple = Fruit()
apple.print_color() # ????

File: fruit.rb
class Fruit
    def initialize(data = {"color" => "unknown"})
        @data = data
    end

    def set_color(color)
        @data["color"] = color
    end

    def print_color
        puts "I am #{@data["color"]}\n"
    end
end


orange = Fruit.new()
orange.set_color("orange")
orange.print_color

apple = Fruit.new()
apple.print_color()